WrightTools/artists/_interact.py

```diff
@@ -0,0 +1,381 @@
+    verbose : boolean (optional)
+        Toggle talkback. Default is True.
+    """
+    # unpack
+    channel = get_channel(data, channel)
+    xaxis, yaxis = get_axes(data, [xaxis, yaxis])
+    cmap = get_colormap(channel)
+    current_state = Bunch()
+    # create figure
+    nsliders = data.ndim - 2
+    if nsliders < 0:
+        print("note enough dimensions")
+        return
+    # TODO: implement aspect; doesn't work currently because of our incorporation of colorbar
+    fig, gs = create_figure(width="single", nrows=7 + nsliders, cols=[1, 1, 1, 1, 1, "cbar"])
+    # create axes
+    ax0 = plt.subplot(gs[1:6, 0:5])
+    ax0.patch.set_facecolor("w")
+    ax0.grid(b=True)
+    cax = plt.subplot(gs[1:6, -1])
+    sp_x = add_sideplot(ax0, "x", pad=0.1)
+    sp_y = add_sideplot(ax0, "y", pad=0.1)
+    ax_local = plt.subplot(gs[0, 0], aspect='equal', frameon=False)  # facecolor='#EEEEEE')
+    ax_title = plt.subplot(gs[0, 3], frameon=False)
+    ax_title.text(0.5, 0.5, data.natural_name, fontsize=18,
+                  horizontalalignment='center', verticalalignment='center',
+                  transform=ax_title.transAxes)
+    ax_title.set_axis_off()
+    # NOTE: there are more axes here for more buttons / widgets in future plans
+    # create lines
+    x_color = "#00BFBF"  # cyan with saturation increased
+    y_color = "coral"
+    line_sp_x = sp_x.plot([None], [None], visible=False, color=x_color)[0]
+    line_sp_y = sp_y.plot([None], [None], visible=False, color=y_color)[0]
+    crosshair_hline = ax0.plot([None], [None], visible=False, color=x_color)[0]
+    crosshair_vline = ax0.plot([None], [None], visible=False, color=y_color)[0]
+    current_state.xpos = crosshair_hline.get_ydata()[0]
+    current_state.ypos = crosshair_vline.get_xdata()[0]
+    # create buttons
+    current_state.local = local
+    radio = RadioButtons(ax_local, (' global', ' local'))
+    if local:
+        radio.set_active(1)
+    else:
+        radio.set_active(0)
+    for circle in radio.circles:
+        circle.set_radius(0.14)
+    # create sliders
+    sliders = {}
+    for axis in data.axes:
+        if axis not in [xaxis, yaxis]:
+            slider_axes = plt.subplot(gs[~len(sliders), :]).axes
+            slider = Slider(
+                slider_axes, axis.label, 0, axis.points.size - 1, valinit=0, valstep=1
+            )
+            sliders[axis.natural_name] = slider
+            slider.ax.vlines(
+                    range(axis.points.size - 1), *slider.ax.get_ylim(),
+                    colors='k', linestyle=':', alpha=0.5
+            )
+    # initial xyz start are from zero indices of additional axes
+    slices = get_slices(sliders, data.axes, verbose=verbose)
+    zi = channel[slices]
+    zi = zi.squeeze()
+    if wt_kit.get_index(data.axes, xaxis) < wt_kit.get_index(data.axes, yaxis):
+        zi = zi.T.copy()
+    current_state.slices = slices
+    # TODO: should we use pcolormesh or pcolor?
```

Comment:
I would lean pcolor, as it is more general, and I'm not sure we can assume even spacing.

Reply:
pcolormesh is faster, though, and speed is important with interactivity, so willing to be convinced otherwise, especially if it impacts user experience.

Reply:
Nothing is tested with pcolor vs. pcolormesh yet, but I will say speed is on my mind, especially with dense 2D datasets such as we will have with ccd images.  We should investigate further, but I vote we leave it for now

Reply:
I can agree to this